File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let netId = 0;
    let networks = Array(n).fill(-1);
    
    let stack = [];
    
    for(let i = 0; i < n; i++) {
        if(networks[i] != -1) continue;
        
        stack.push(i);
        networks[i] = ++netId;
        while(stack.length > 0) {
            let cur = stack.pop();
            computers[cur].forEach((linked, c) => {
                if(linked && networks[c] == -1) {
                    stack.push(c);
                    networks[c] = netId
                }
            })
        }
    }
    
    answer = netId
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        List<List<Integer>> adj = new ArrayList<>();
        for(int i = 0; i < computers.length; i++) {
            adj.add(i, new ArrayList<>());
            for(int j = 0; j < computers[i].length; j++) {
                if(i != j && computers[i][j] == 1) adj.get(i).add(j);
            }
        }
        
        Set<Integer> s = new HashSet<>();
        int answer = 0;
        
        for(int i = 0; i < computers.length; i++) {
            if(!s.contains(i)) {
                answer++;
                dfs(adj, s, i);
            }
        }
        
        
        return answer;
    }
    
    public void dfs(List<List<Integer>> adj, Set<Integer> visited, int cur) {
        visited.add(cur);
        for(int next : adj.get(cur)) {
            if(!visited.contains(next)) {
                dfs(adj, visited, next);
            }
        }
    }
}